### PR TITLE
Update Ruby to 3.3.6

### PR DESCRIPTION
**Pull request by Uaitt**
As the title says :)

### CODE REVIEW

**Review by simi — APPROVED** [review]

**Comment by simi** [comment]
Thanks for upgrade @Uaitt! Once CI passes I can merge and deploy.

File: .github/actions/setup-rubygems.org/action.yml [hunk 1]
```diff
@@ -23,7 +23,7 @@ runs:
       if: github.secret_source != 'None' && inputs.install-avo-pro == 'true'
       run: |
         printf "BUNDLE_WITH=avo\nRAILS_GROUPS=avo\n" >> $GITHUB_ENV
-    - uses: ruby/setup-ruby@52753b7da854d5c07df37391a986c76ab4615999 # v1.191.0
+    - uses: ruby/setup-ruby@c6d4d2ab97f832d0fea49200f14c2441e56fd0a6 # v1.200.0
```
Comment (martinemde) [inline .github/actions/setup-rubygems.org/action.yml:26]: The [released](https://github.com/ruby/setup-ruby/releases/tag/v1.200.0) commit for 1.200.0 is `217c988b8c2bf2bacb2d5c78a7e7b18f8c34daed`

Could we use that instead?

**Review by martinemde — APPROVED** [review]

**Comment by Uaitt** [comment]
@simi and @martinemde I have rebased this branch so that it's up to date and I have changed the commit hash of the `ruby/setup-ruby` action

**Review by martinemde — APPROVED** [review]

**Comment by Uaitt** [comment]
Any idea why the coverage is getting lower with this commit? 😅

**Comment by simi** [comment]
@Uaitt not really, I think worth to explore. I can take a look.

**Comment by martinemde** [comment]
When I looked it seems like the Avo tests were not being counted. All of the coverage changes were due to avo related files.

I'm going to go ahead and merge this now. I'm don't think coverage actually changed because of this.

**Comment by simi** [comment]
@martinemde But per my understanding Avo tests were triggered in this PR also. Those strange coverage drops often reveal codepath changes. :thinking:

**Comment by simi** [comment]
I thought all tests are running for all PRs. But maybe I'm wrong, since AFAIK secrets are not provided to community PRs. :thinking: 

image

**Pull request merged**